homeassistant/components/transmission/__init__.py

```diff
@@ -163,6 +172,45 @@ async def async_setup(self) -> None:
 
         self._tm_data = TransmissionData(self.hass, self.config_entry, self.tm_api)
 
+        @callback
+        def update_unique_id(
+            entity_entry: entity_registry.RegistryEntry,
+        ) -> dict[str, Any] | None:
+            """Update unique ID of entity entry."""
+
+            name_to_key = {
+                # Sensors
+                "Down Speed": "download",
+                "Up Speed": "upload",
+                "Status": "status",
+                "Active Torrents": "active_torrents",
+                "Paused Torrents": "paused_torrents",
+                "Total Torrents": "total_torrents",
+                "Completed Torrents": "completed_torrents",
+                "Started Torrents": "started_torrents",
+                # Switches
+                "Switch": "on_off",
+                "Turtle Mode": "turtle_mode",
+            }
+
+            match = re.search(
+                f"{self.api.host}-{self.config_entry.data[CONF_NAME]} (?P<name>.*)",
+                entity_entry.unique_id,
+            )
+
+            if match is None:
+                return None
+
+            if (name := match.group("name")) in name_to_key:
+                return {
+                    "new_unique_id": f"{self.config_entry.entry_id}-{name_to_key[name]}"
+                }
+            return None
```

Comment:
This should either find a match and update or not. Can this be re-written so that we don't get 3 returns?